compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/symbols/Symbol.java

```diff
@@ -35,6 +35,13 @@
      */
     Optional<String> getName();
 
+    /**
+     * Retrieves the original name of the symbol if it is associated with an identifier.
+     *
+     * @return The name of the symbol if applicable
+     */
+    Optional<String> getOriginalName();
```

Comment:
What's the need for this? IIRC, the original discussion was to change `getName()` to return the original name 🤔

Reply:
In lang libs, Since we are changing the `getName()` to return the unquoted names for quoted names (Ex: `'map()`), we may ended up having same [error](https://github.com/ballerina-platform/ballerina-lang/issues/30242) for `'map()`. 

Please try this example
```ballerina
public function main() {
    string[] hello = ["a", "b"];
    int[] mapResult = hello.'map(function (string a) returns int {
        return 12;
    }).<cursor>
}
```

We need two APIs to retrieve the original name (with the quote) and the changed name (which we get from `getName()`